app/models/legal_document.rb

```diff
@@ -69,50 +70,57 @@ def self.remove_duplicates
     start = Time.zone.now.to_f
     Rails.logger.info '-----> Removing legal documents duplicates'
     count = 0
-    modified = Array.new
+    modified = []
 
-    LegalDocument.where(documentable_type: "Domain").where.not(checksum: [nil, ""]).find_each do |orig_legal|
+    LegalDocument.where(documentable_type: 'Domain')
+                 .where.not(checksum: [nil, ''])
+                 .find_each do |orig_legal|
       next if modified.include?(orig_legal.checksum)
-      next if !File.exist?(orig_legal.path)
+      next unless File.exist?(orig_legal.path)
+
       modified.push(orig_legal.checksum)
 
-      LegalDocument.where(documentable_type: "Domain", documentable_id: orig_legal.documentable_id).
-          where(checksum: orig_legal.checksum).
-          where.not(id: orig_legal.id).where.not(path: orig_legal.path).each do |new_legal|
-        unless modified.include?(orig_legal.id)
-          File.delete(new_legal.path) if File.exist?(new_legal.path)
-          new_legal.update(path: orig_legal.path)
-          count += 1
-          Rails.logger.info "File #{new_legal.path} has been removed by Domain "\
-                            "#{new_legal.documentable_id}. Document id: #{new_legal.id}"
-        end
+      LegalDocument.where(documentable_type: 'Domain', documentable_id: orig_legal.documentable_id)
+                   .where(checksum: orig_legal.checksum)
+                   .where.not(id: orig_legal.id)
+                   .where.not(path: orig_legal.path).each do |new_legal|
```

Comment:
Same here, why not put path: `orig_legal.path` in the query above? The problem with `where` is that it won't work as a filter in the ruby concept. When `where` is called, it will always query the database, even with the sample that was returned from the database. Even if you use `includes`.